libcloud/http.py

```diff
@@ -241,8 +241,6 @@ def prepared_request(self, method, url, body=None,
 
         prepped = self.session.prepare_request(req)
 
-        prepped.body = body
```

Comment:
I was wondering where we actually then send a body when a prepared request is used and it looks like it's done here - https://github.com/apache/libcloud/pull/1488/files#diff-445eb619a1749349909b5157c751c3a2R239.

I think it would still be good to add some test cases for all this various functionality.

It may not be a bad idea to add a test case which uses a mock http server and tests all the various scenarios (I might look into that if I get a chance).